From 1cde2cae0b4483c3cbde6e2d1aae60470542fd6a Mon Sep 17 00:00:00 2001
From: henning <>
Date: Wed, 22 Apr 2026 13:54:50 +0000
Subject: [PATCH 05/30] in show_peer_msg, grow the buffer to hold the sprintf'd
 stratum by 1 byte. the previous buffer was large enough since startum is
 clamped to 0..15, however, it is a bit much to ask for analyzers - including
 those in compilers - to detect that, an extra byte on the stack costs us
 effectively nothing, and it feels a bit more robust. triggered by bcook's
 portable diffs, ok claudio

---
 src/usr.sbin/ntpd/ntpd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/usr.sbin/ntpd/ntpd.c b/src/usr.sbin/ntpd/ntpd.c
index ccc2c12509b..46a38aba8e0 100644
--- a/src/usr.sbin/ntpd/ntpd.c
+++ b/src/usr.sbin/ntpd/ntpd.c
@@ -1,4 +1,4 @@
-/*	$OpenBSD: ntpd.c,v 1.144 2026/04/21 14:36:00 henning Exp $ */
+/*	$OpenBSD: ntpd.c,v 1.145 2026/04/22 13:54:50 henning Exp $ */
 
 /*
  * Copyright (c) 2003, 2004 Henning Brauer <henning@openbsd.org>
@@ -839,7 +839,7 @@ show_peer_msg(struct imsg *imsg, int calledfromshowall)
 {
 	struct ctl_show_peer	*cpeer;
 	int			 cnt;
-	char			 stratum[3];
+	char			 stratum[4];
 	static int		 firsttime = 1;
 
 	if (imsg->hdr.type == IMSG_CTL_SHOW_PEERS_END) {
-- 
2.53.0

